apps/web/components/booking/pages/BookingPage.tsx

```diff
@@ -77,22 +56,112 @@ import { HashLinkPageProps } from "../../../pages/d/[link]/book";
 import { TeamBookingPageProps } from "../../../pages/team/[slug]/book";
 
 type BookingPageProps = BookPageProps | TeamBookingPageProps | HashLinkPageProps;
+const BookingFields = ({
+  fields,
+  locations,
+  rescheduleUid,
+  isDynamicGroupBooking,
+}: {
+  fields: BookingPageProps["eventType"]["bookingFields"];
```

Comment:
I have a counter reasoning here. I am just passing on a prop created by getSSRProps here. If I end up generating a different variation of FieldsType in getSSRProps, the typecheck will pass, but it might crash at runtime.

Reply:
That's a problem of type signature differing from the runtime signature. It doesn't matter where this comes from as long as signature matches then there shouldn't be any problems.